### ISSUE TITLE
[Package Request] - OpenDKIM

### ISSUE THREAD

**Issue by makmiller-clgx** [amazonlinux/amazon-linux-2023#337]
**What package is missing from Amazon Linux 2023? Please describe and include package name.**
Package Name: OpenDKIM

Tried: yum --showduplicates list *dkim* 
Error: No matching Packages to list

**Is this an update to the existing package or a new package request?**
I do not see the package list in available packages nor do I see it in unsupported.

**Is this package available in Amazon Linux 2? If it is available via external sources such as EPEL, please specify.**
Yes, OpenDKIM is available in AL 2. It is available via EPEL. However, AL2023 does not support EPEL

Tried: yum --showduplicates list *epel* 
Error: No matching Packages to list

**Any additional information you'd like to include? (use-cases, etc)**

**Comment by kivengh** [comment]
Do you have any plan to include OpenDKIM in Amazon Linux ?

**Comment by ozbenh** [comment]
This is under investigation

**Comment by Bendolfc** [comment]
Is there any update on this, DKIM is pretty important these days.

**Comment by daniejstriata** [comment]
I was able to build `opendkim`, from the latest Fedora release, using COPR in my repo here:
https://copr.fedorainfracloud.org/coprs/faramirza/al2023/package/opendkim/
You would need to build the following dependencies before you can build `opendkim`:
```
firebird
freetds
sqlite2
opendbx
```

**Comment by tonyswu** [comment]
It's been more than a year since the creation of this request. Any update on this?

**Comment by alsunseri** [comment]
I also would like to know the status of the "investigation."
Its one thing to decide to force users to switch from cron to systemd timers or to force users to switch from rsyslog to journalctl. 
But there is no replacement for opendkim, the package has simply been removed. 
Why was this done? When will this oversight be remediated? 
Or was this intentional?

**Comment by ozbenh** [comment]
It wasn't removed per-se, Amazon Linux never carried it. Keep in mind that EPEL is fundamentally unsupported software.

There are some concerns about the risk and dependencies but it's still in the list of potential adds.

As for "forcing users to switch from cron etc...." this is a rather unfair comment as this is a general trend that has happened progressively accross all Linux distributions over the last decade

**Comment by alsunseri** [comment]
> It wasn't removed per-se, Amazon Linux never carried it. Keep in mind that EPEL is fundamentally unsupported software.

ok fair enough, i forgot that was an epel package.  The deeper question is what can AL2023 users do if they need to continue to send emails ? I have been sending emails via postfix on amazon for a decade without incident. 
For more than one client. 
Can Amazon pick up this package or should we try to compile opendkim ? Or should we just keep using AL2 ?

**Comment by ozbenh** [comment]
I am trying to get some attention internally on this, but in the meantime, you can use your own build (or the above COPR one) if that works for you.

**Comment by ryu1** [comment]
Starting in February 2024, Google will be updating its guidelines for sending email and recommends setting up SPF, DKIM, DMARC, and ARC.
I strongly feel the need for opendkim.
However, there is also little information on how to make and install from its source code.

**Comment by kivengh** [comment]
Be careful if you plan to use SSSD with OpenDKIM in Amazon Linux 2023. There is some issue using sssd 2.9+ and opendkim : https://github.com/trusteddomainproject/OpenDKIM/issues/198

**Comment by pll** [comment]
Still waiting for an answer on OpenDKIM...

**Comment by btpence** [comment]
I was finally able to compile opendkim on Amazon Linux 2023.  It didn't
work out of the box and required a little fiddling.

If anyone is interested, contact me directly....
***@***.***

Brian Pence
Celestial Software
901-283-1970
http://www.celestialsoftware.net

**Comment by btpence** [comment]
Yes.  I'll post the complete lists of steps somewhere....  I may have to
reconstruct how to do it on a new vanilla instance.
Brian Pence
Celestial Software
901-283-1970
http://www.celestialsoftware.net

**Comment by btpence** [comment]
I've summarized the approach here.  Please try following these instructions
and let me know if it works.

https://www.celestialsoftware.net/compiling-opendkim-on-amazon-linux-2023/


Brian Pence
Celestial Software
901-283-1970
http://www.celestialsoftware.net

**Comment by stewartsmith** [comment]
OpenDKIM has been added as part of the [2023.6.20250303](https://docs.aws.amazon.com/linux/al2023/release-notes/relnotes-2023.6.20250303.html) release.
